### PR TITLE
On Node Properties

### PR DESCRIPTION
**Pull request by deltakosh**
Boolean, lists ,float and float-slider are supported

image

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15770/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15770/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15770/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15770/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15770/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15770/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15770/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15770/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15770/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15770/merge/testResults/webgpuplaywright/index.html

File: packages/tools/nodeRenderGraphEditor/src/graphSystem/properties/genericNodePropertyComponent.tsx [hunk 1]
```diff
@@ -11,8 +11,9 @@ import { SliderLineComponent } from "shared-ui-components/lines/sliderLineCompon
 import { Color4LineComponent } from "shared-ui-components/lines/color4LineComponent";
 import type { NodeRenderGraphBlock } from "core/FrameGraph/Node/nodeRenderGraphBlock";
 import type { IEditablePropertyListOption } from "core/Decorators/nodeDecorator";
-import { PropertyTypeForEdition, type IEditablePropertyOption, type IPropertyDescriptionForEdition } from "core/Decorators/nodeDecorator";
+import { PropertyTypeForEdition, type IPropertyDescriptionForEdition } from "core/Decorators/nodeDecorator";
```
Comment (Popov72) [inline packages/tools/nodeRenderGraphEditor/src/graphSystem/properties/genericNodePropertyComponent.tsx:14]:
```suggestion
import type { IPropertyDescriptionForEdition } from "core/Decorators/nodeDecorator";
import { PropertyTypeForEdition } from "core/Decorators/nodeDecorator";
```

**Comment by deltakosh** [comment]
I replicated Blender behavior

File: packages/tools/nodeEditor/src/components/propertyTab/propertyTabComponent.tsx [hunk 2]
```diff
@@ -81,13 +82,18 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
             }
         });
 
+        this._onForceUpdateObserver = this.props.globalState.stateManager.onForceUpdateObservable.add(() => {
```
Comment (alexchuber) [inline packages/tools/nodeEditor/src/components/propertyTab/propertyTabComponent.tsx:85]: nit: Maybe the onForceUpdate name could be more specific? Something like onForceUpdateProperties? Just so it's super clear what part of the app is being force updated :)

File: packages/dev/sharedUiComponents/src/nodeGraphSystem/automaticProperties.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Function used to force a rebuild of the node system
+ * @param source source object
+ * @param stateManager defines the state manager to use
+ * @param propertyName name of the property that has been changed
+ * @param notifiers list of notifiers to use
+ * @param forceUpdate defines if we should force a react update
+ */
+export function ForceRebuild(source: any, stateManager: StateManager, propertyName: string, notifiers?: IEditablePropertyOption["notifiers"], forceUpdate = false) {
```
Comment (alexchuber) [inline packages/dev/sharedUiComponents/src/nodeGraphSystem/automaticProperties.ts:12]: nit: if no other exported functions are going to be in this file, could the file be named after its function (ForceRebuild)?

**Review by alexchuber — APPROVED** [review]
Love the idea of embedded properties in the nodes, and also very happy to see the DRYness!

File: packages/dev/sharedUiComponents/src/nodeGraphSystem/automaticProperties.ts (hunk 3 above)
Comment (deltakosh) [reply packages/dev/sharedUiComponents/src/nodeGraphSystem/automaticProperties.ts:12]: Good take but I plan to add more ;)

File: packages/tools/nodeEditor/src/components/propertyTab/propertyTabComponent.tsx (hunk 2 above)
Comment (deltakosh) [reply packages/tools/nodeEditor/src/components/propertyTab/propertyTabComponent.tsx:85]: Good point! I'll fix that

**Comment by deltakosh** [comment]
image

**Review by Popov72 — APPROVED** [review]

**Pull request merged**